### PR TITLE
Adds QC Report Generation

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

#203 
#205 

## Purpose/Implementation Notes

Implements the usage of FastQC and MultiQC to the Salmon pipelines.

Also adds a table association fix to the Salmon pipe.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Functional tests
New tests are provided with lots of new assertions.

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots

screen shot 2018-05-23 at 4 01 21 pm

### CODE REVIEW

File: format_nomad_with_env.sh [hunk 1]
```diff
@@ -167,6 +167,19 @@ export_log_conf (){
...
+        logging {
+          type = \"awslogs\"
+          config {
+            awslogs-region = \"$REGION\",
+            awslogs-group = \"data-refinery-log-group-$USER-$STAGE\",
+            awslogs-stream = \"log-stream-$1-docker-$USER-$STAGE\"
+          }
+        }
+"
+}
```
Comment (kurtwheeler) [inline format_nomad_with_env.sh]: Are you sure you meant to add this? It seems like the block of code immediately preceding this already does this.

File: workers/data_refinery_workers/processors/salmon.py [hunk 2]
```diff
@@ -266,11 +352,16 @@ def _run_fastqc(job_context: Dict) -> Dict:
                 formatted_command,
                 processor_job=job_context["job_id"])
 
+    start_time = timezone.now()
     completed_command = subprocess.run(formatted_command.split(),
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE)
+    end_time = timezone.now()
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/salmon.py]: `start_time` and `end_time` don't seem to be used anywhere.

File: workers/data_refinery_workers/processors/test_salmon.py [hunk 3]
```diff
@@ -98,11 +98,44 @@ def test_salmon(self):
...
+        self.assertTrue(win['success'])
+        win = salmon._run_multiqc(win_context)
+        self.assertTrue(win['success'])
+
+        for file in win['qc_files']:
+            self.assertTrue(os.path.isfile(file.absolute_file_path))
+
+        fail_context = {
+            'job': job,
+            'job_id': 'hippityhoppity',
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/test_salmon.py]: :joy:

File: workers/data_refinery_workers/processors/test_salmon.py [hunk 4]
```diff
@@ -98,11 +98,44 @@ def test_salmon(self):
...
+            'qc_directory': "/home/user/data_store/raw/TEST/SALMON/derp",
+            'original_files': [],
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/test_salmon.py]: Is this empty list what causes this to fail?

Comment (Miserlou) [reply workers/data_refinery_workers/processors/test_salmon.py]: Yes, fastqc will fail without valid input, so this test checks the fail path appropriate sets the `success` keyvalue

**Pull request merged**